spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/LocalizedQueueConnectionFactory.java

```diff
@@ -398,7 +398,7 @@ default ConnectionFactory locate(String[] adminUris, Map<String, String> nodeToA
 						String node = (String) queueInfo.get("node");
 						if (node != null) {
 							String nodeUri = nodeToAddress.get(node);
-							if (uri != null) {
+							if (nodeUri != null) {
```

Comment:
I think only this line change is a back-port candidate. 😄

Reply:
Agreed plus maybe the NodeLocatorTest.